src/Raven.Server/Documents/ResourceCache.cs

```diff
@@ -210,7 +210,7 @@ public IDisposable RemoveLockAndReturn(string databaseName, Action<TResource> on
                 });
             }
 
-            if (current.IsFaulted && current.Exception?.Data.Contains(DatabasesLandlord.DoNotRemove) == false)
+            if (current.IsFaulted && current.Exception?.InnerException?.Data.Contains(DatabasesLandlord.DoNotRemove) == false)
```

Comment:
This is the fix?
Wonder if we shouldn't do something a bit more robust... 
In any case other places seems to use `ExtractSingleInnerException()`

Reply:
Yes, I don't like it. The problem is that we fail to get the lock on the db, but then we try to remove it anyway?
We can handle that scenario specifically, and this looks ripe for a small change in the error handling killing us

Reply:
now we reuse the same static method is all places that check if the database is locked